### ISSUE TITLE
Improve ETA estimate during RETH node sync

### ISSUE THREAD

**Issue by liangjh** [paradigmxyz/reth#5840]
### Describe the bug

The ETA estimates during the initiall sync of the node vary wildly.  Wonder if they are more of a distraction than useful - how are the ETAs being calculated?   Request is to either improve the estimate, or consider removing altogether for some stages that are particularly lengthy.

From sample of node logs below, the ETA estimate varied significantly across checkpoints.   6 days => 1 month 12 days => 18 days => 1 month 11 days 

Hopefully the sync doesn't really take near the high end of the estimate, or I may have to cancel this endeavor this altogether  :)

### Steps to reproduce

Run RETH node.  In my case I set debug.tip to a high block number (i.e. 18M), but I imagine this issue would manifest itself in any case.   I've noticed this being more acute in stage 5 (Execution) but likely endemic to all stages.

### Node logs

```text
Sample of node logs.  The below is on stage 5 (Execution), where this problem is acute; but we need to 

...
2023-12-20T22:39:00.187247Z  INFO reth::node::events: Stage committed progress pipeline_stages=5/13 stage=Execution checkpoint=4587893 target=18000000 progress=2.55% eta=6days 18h 17m 58s
...
2023-12-21T10:47:25.915809Z  INFO reth::node::events: Stage committed progress pipeline_stages=5/13 stage=Execution checkpoint=5018665 target=18000000 progress=4.39% eta=1month 12days 5h 49m 31s
...
2023-12-21T11:44:01.634690Z  INFO reth::node::events: Stage committed progress pipeline_stages=5/13 stage=Execution checkpoint=5060006 target=18000000 progress=4.57% eta=18days 23h 51m 46s
...
2023-12-21T11:44:13.943164Z  INFO reth::cli: Status connected_peers=95 freelist=202691 stage=Execution checkpoint=5060006 target=18000000 progress=4.57% eta=18days 23h 51m 34s
...
2023-12-21T14:50:56.302999Z  INFO reth::node::events: Stage committed progress pipeline_stages=5/13 stage=Execution checkpoint=5141503 target=18000000 progress=4.91% eta=1month 11days 19h 27m 14s
```

### Platform(s)

Mac (Apple Silicon)

### What version/commit are you on?

0.1.0-alpha.13

### What database version are you on?

1

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

N/A (default reth.toml config)

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by liangjh** [comment]
i could be reading this incorrectly, but it looks like we're just using information from the most recent checkpoint to calc the elapsed time.   could we improve by utilizing all information accumulated so far?   i.e. total processed and total elapsed time?   (or a more complex one would be a rolling number i.e. last N checkpoints)?   the idea is that the estimate can be smoothed out given more info.  if only the latest is used that may explain why the estimates are volatile.

https://github.com/paradigmxyz/reth/blob/b37cd833a43507fc8a1bb6c88fbdff235a4d3585/bin/reth/src/node/events.rs#L394

lmk what you think.  a simple revision to test out could be to utilize totals.

**Comment by shekhirin** [comment]
Agreed, we're disabling the ETA for Execution stage in https://github.com/paradigmxyz/reth/pull/5839, so it doesn't confuse people. Need to revisit the logic in the future, taking into account more than two points in time and maybe a projection of execution performance for historical blocks.

**Comment by joshieDo** [comment]
related https://github.com/paradigmxyz/reth/issues/6741
